toonz/sources/tnztools/tooloptions.cpp

```diff
@@ -2883,6 +2849,19 @@ void ToolOptions::hideEvent(QShowEvent *) {
 
 //-----------------------------------------------------------------------------
 
+void ToolOptions::onToolOptionsBoxChanged() {
+  TTool *tool = TTool::getApplication()->getCurrentTool()->getTool();
+  std::map<TTool *, ToolOptionsBox *>::iterator it = m_panels.find(tool);
+  if (it != m_panels.end()) {
+    ToolOptionsBox *panel = it->second;
+    m_panels.erase(it);
+    layout()->removeWidget(panel);
```

Comment:
@blackwarthog 
Why did you have to delete and recreate the tool options each time you changed the style? This appears to be a time consuming operation.
Is it not enough to hide unnecessary options with `filterControls()` ?

Reply:
Sorry for late answer. Yes, possibly it was not good idea... I've made this code before mypaint brushes was implemented for toonz raster level (it was implemented in parallel)